### PR TITLE
Add ModelPool::setFieldNames

### PR DESCRIPTION
**Pull request by josephbirkner**
Motivation: Allow two features which are encoded with different field dicts to be evaluated under one OverlayNode, which combines the two feature nodes which come from different models.

### CODE REVIEW

File: src/model/fields.cpp [hunk 1]
```diff
@@ -26,6 +24,16 @@ Fields::Fields() {
     addStaticKey(Elevation, "elevation");
 }
 
+Fields::Fields(const Fields& other) :
```
Comment (josephbirkner) [inline src/model/fields.cpp:27]: Note: This copy ctor impl cannot be defaulted because of the shared_lock member.

**Review by johannes-wolf — CHANGES_REQUESTED** [review]

File: src/model/model.cpp [hunk 2]
```diff
@@ -428,4 +428,22 @@ void ModelPool::read(std::istream& inputStream) {
     }
 }
 
+std::shared_ptr<simfil::Fields> ModelPool::takeFieldsDictOwnership()
+{
+    impl_->fieldNames_ = std::make_shared<Fields>(*impl_->fieldNames_);
+    return impl_->fieldNames_;
+}
+
+void ModelPool::transcode(std::shared_ptr<simfil::Fields> const& newDict)
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply src/model/model.cpp]: Good point. As we have `fieldNames()`, I went with "setFieldNames".

File: src/model/model.cpp [hunk 3]
```diff
@@ -428,4 +428,22 @@ void ModelPool::read(std::istream& inputStream) {
     }
 }
 
+std::shared_ptr<simfil::Fields> ModelPool::takeFieldsDictOwnership()
+{
+    impl_->fieldNames_ = std::make_shared<Fields>(*impl_->fieldNames_);
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply src/model/model.cpp]: I got rid of the method.

**Review by johannes-wolf — APPROVED** [review]
LGTM :guitar:

**Pull request merged**